src/main/java/com/minecolonies/coremod/commands/ActionMenu.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    @Override
+    public ActionMenuType getMenuType()
+    {
+        return (ActionMenuType) menuType;
+    }
+
+    public String getDescription()
+    {
+        return description;
+    }
+
+    public ForgePermissionNodes getForgePermissionNode()
+    {
+        return forgePermissionNode;
+    }
+
+    public Class<? extends IActionCommand> getActionCommandClass()
+    {
+        return actionCommandClass;
+    }
+
+    private List<ActionArgument> getAllArgumentsList()
+    {
+        return getAllArgumentsList(getActionArgumentList());
+    }
+
+    private List<ActionArgument> getAllArgumentsList(@NotNull final List<ActionArgument> currentActionArgumentList)
+    {
+        final List<ActionArgument> allActionArgumentsList = new ArrayList<>();
+        for (final ActionArgument actionArgument : currentActionArgumentList)
+        {
+            allActionArgumentsList.add(actionArgument);
+            if (actionArgument.isValueSet())
+            {
+                allActionArgumentsList.addAll(getAllArgumentsList(actionArgument.getActionArgumentList()));
+            }
+        }
+        return allActionArgumentsList;
+    }
+
+    public Colony getColonyForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.COLONY == actionArgument.getType())
+                {
+                    return (Colony) actionArgument.getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    public EntityPlayerMP getPlayerForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.PLAYER == actionArgument.getType())
+                {
+                    return (EntityPlayerMP) actionArgument.getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    public CitizenData getCitizenForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.CITIZEN == actionArgument.getType())
+                {
+                    return (CitizenData) actionArgument.getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    public Boolean getBooleanForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.BOOLEAN == actionArgument.getType())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)